src/theory/arith/callbacks.cpp

```diff
@@ -87,37 +87,25 @@ void FarkasConflictBuilder::reset(){
   d_consequent = NullConstraint;
   d_constraints.clear();
   d_consequentSet = false;
-  PROOF(d_farkas.clear());
   Assert(!underConstruction());
 }
 
 /* Adds a constraint to the constraint under construction. */
 void FarkasConflictBuilder::addConstraint(ConstraintCP c, const Rational& fc){
-  Assert(
-      !PROOF_ON()
-      || (!underConstruction() && d_constraints.empty() && d_farkas.empty())
-      || (underConstruction() && d_constraints.size() + 1 == d_farkas.size()));
-  Assert(PROOF_ON() || d_farkas.empty());
+  Assert(d_farkas.empty());
```

Comment:
```suggestion
  Assert((!underConstruction() && d_constraints.empty() && d_farkas.empty())
      || (underConstruction() && d_constraints.size() + 1 == d_farkas.size()));
```

Same as above, we should treat proofs as "always on", but otherwise not modify the core of how arith computes and stores farkas coefficients.